### PR TITLE
test-devlink-encap.sh: Use sysfs instead of devlink on backport

### PR DESCRIPTION
**Pull request by ZiyadAtiyyeh**
Signed-off-by: Ziyad Atiyyeh <ziyadat@mellanox.com>

### CODE REVIEW

**Comment by roidayan** [comment]
Should be lower case devlink

File: test-devlink-encap.sh [hunk 1]
```diff
@@ -61,7 +81,11 @@ start_check_syndrome
 set_encap disable
 test_encap disable
 switch_mode_legacy
-extra_mode="encap enable"
+if [ "$devlink_compat" = 1 ]; then
+    set_encap enable
+else
+    extra_mode="encap enable"
+fi
```
Comment (roidayan) [inline test-devlink-encap.sh]: Many tests use extra mode or just this one?
Maybe we should disable this option and just always call to enable/disable encap.

File: test-devlink-encap.sh [hunk 2]
```diff
@@ -11,15 +11,35 @@ my_dir="$(dirname "$0")"
...
+    if [ "$devlink_compat" = 1 ]; then
+	if [ "$val" == "disable" ]; then
+            val="none"
+        else
+	    val="basic"
+        fi
+        echo $val > /sys/kernel/debug/mlx5/$PCI/compat/encap && success || fail "Failed to set encap"
+    else
+        devlink dev eswitch set pci/$PCI encap $val && success || fail "Failed to set encap"
+    fi
```
Comment (roidayan) [inline test-devlink-encap.sh]: This should probably mode to common like other devlink options e.g.inline multipath etc

Comment (roidayan) [reply test-devlink-encap.sh]: move*

File: test-devlink-encap.sh (hunk 1 above)
Comment (ZiyadAtiyyeh) [reply test-devlink-encap.sh]: I only faced this issue with this test. So we can keep it as is for now.

File: test-devlink-encap.sh [hunk 3]
```diff
@@ -11,15 +11,35 @@ my_dir="$(dirname "$0")"
 not_relevant_for_cx5
 
 function get_encap() {
-    output=`devlink dev eswitch show pci/$PCI`
+    if [ "$devlink_compat" = 1 ]; then
+        output=`cat /sys/kernel/debug/mlx5/$PCI/compat/encap`
+        if [ "$output" == "none" ]; then
+            encap="disable"
+        else
```
[the remark this answers is not in the document]
Comment (ZiyadAtiyyeh) [reply test-devlink-encap.sh]: So, I can check also the other value "basic" with if else then the else will disply error if it is not "none" and "basic".

what do you think?

File: test-devlink-encap.sh (hunk 2 above)
Comment (ZiyadAtiyyeh) [reply test-devlink-encap.sh]: There will be one call for this function if we moved it to the common so we can keep it. what do you think?

Comment (roidayan) [reply test-devlink-encap.sh]: Ok

File: test-devlink-encap.sh (hunk 1 above)
Comment (roidayan) [reply test-devlink-encap.sh]: Ok

File: test-devlink-encap.sh (hunk 3 above)
[the remark this answers is not in the document]
Comment (roidayan) [reply test-devlink-encap.sh]: Yes

File: test-devlink-encap.sh (hunk 1 above)
Comment (ZiyadAtiyyeh) [reply test-devlink-encap.sh]: Ack.

File: test-devlink-encap.sh (hunk 2 above)
Comment (ZiyadAtiyyeh) [reply test-devlink-encap.sh]: Ack.

**Comment by ZiyadAtiyyeh** [comment]
Suggested fixes applied.

File: test-devlink-encap.sh (hunk 3 above)
[the remark this answers is not in the document]
Comment (ZiyadAtiyyeh) [reply test-devlink-encap.sh]: Fixed.

**Pull request merged**